pubnub/lib/src/dx/_endpoints/history.dart

```diff
@@ -154,25 +154,46 @@ class BatchHistoryResultEntry {
   /// Otherwise, it will be `null`.
   Map<String, dynamic>? meta;
 
+  /// Error message. this will contain information if message decryption is failed
+  /// for given `message`.
+  String? error;
+
   BatchHistoryResultEntry._(this.message, this.timetoken, this.uuid,
-      this.messageType, this.actions, this.meta);
+      this.messageType, this.actions, this.meta, this.error);
 
   /// @nodoc
   factory BatchHistoryResultEntry.fromJson(Map<String, dynamic> object,
       {CipherKey? cipherKey, Function? decryptFunction}) {
+    var message;
+    String? errorMessage;
+    if (cipherKey == null && decryptFunction is decryptWithKey) {
+      message = object['message'];
+    } else {
+      try {
+        message = decryptFunction is decryptWithKey
```

Comment:
Btw, what's the logic behind it?

Reply:
Ok, now I see that you could pass `decrypt(...)` or `decryptWithKey(...)` function from `ICryptoModule`. Perhaps there's no way to split deserialization and decryption into two separate concerns, this would require significant changes. What I mean by this is parsing the objects as they came without introducing decryption there. This could be a separate layer that takes `BatchHistoryResultEntry` or whatever lightweight DAO object created in the previous step, decrypts its payload and returns decrypted result back to the caller. It's not possible to do it right now, but I wanted to mention about that :)

Reply:
Agreed!
Need to deal with this till the time we completely stop supporting explicit cipherKey passing mechanism.